examples/mgui_with_threadpoolexec.py

```diff
@@ -1,10 +1,12 @@
 """An example of calling a threaded function from a magicgui dock_widget.
 Note: this example requires python >= 3.9
 """
-from magicgui import magic_factory, widgets
-from skimage import feature
+from concurrent.futures import Future, ThreadPoolExecutor
+
+from magicgui import magic_factory
+from skimage import data
+from skimage.feature import blob_log
 from typing_extensions import Annotated
-from concurrent.futures import Future
 
```

Comment:
```suggestion
from typing import Awaitable
```

Reply:
the docstring at the top of the example states that this example requires python 3.9 (in #2581 we had this discussion and resigned ourselves to supporting only 3.9 for this little sliver of functionality for now).
Awaitable is not quite right, since we _are_ depending on at least part of the Futures api here, specifically, the object they return _must_ have an `add_done_callback` and `result` method or it won't work, and Awaitable doesn't provide that.   now... we could "cheat" a bit and accept Awaitable, simply because it's available in typing, but that seems to open the doors to more bugs if the object they return isn't actual a Future?